File: src/breslin/clientside/application/states/applicationMainPartido.cpp
//header
#include "applicationMainPartido.h"

//log
#include "../../tdreamsock/dreamSockLog.h"

//application
#include "../applicationPartido.h"

//game
#include "../../game/gamePartido.h"

//state machine
#include "../../../statemachine/stateMachine.h"

/***************************************
*	CONSTRUCTORS
***************************************/
ApplicationMainPartido::ApplicationMainPartido(ApplicationPartido* applicationPartido) : ApplicationMain(applicationPartido)
{
	mApplicationPartido = applicationPartido;
}

ApplicationMainPartido::~ApplicationMainPartido()
{

}

void ApplicationMainPartido::enter()
{
	ApplicationMain::enter();	
}

void ApplicationMainPartido::execute()
{
	ApplicationMain::execute();	

	if (mApplicationPartido->mButtonHit == mApplicationPartido->mButtonJoinGameB)
	{
		LogString("ApplicationMainPartido::execute....button hit for b"); 
		mApplicationPartido->mButtonHit = NULL;
 		mApplicationPartido->setGame(new GamePartido(mApplicationPartido));
 		mApplicationPartido->getGame()->createStates();
 		mApplicationPartido->getGame()->setStates();
		mApplicationPartido->sendJoinGame(2);
		mApplicationPartido->mStateMachine->changeState(mApplicationPartido->mApplicationPlay);
	}
}

void ApplicationMainPartido::exit()
{
	ApplicationMain::exit();	
}

File: src/breslin/clientside/application/states/application_main_partido.php
var ApplicationMainPartido = new Class(
{

Extends: ApplicationMain,
	
initialize: function(applicationPartido)
{
	this.parent(applicationPartido);

	this.mApplicationPartido = applicationPartido;
},

enter: function()
{
	this.parent();
},

execute: function()
{
	this.parent();

	//join game B
	if (this.mApplicationPartido.mButtonHit == this.mApplicationPartido.mButtonJoinGameB)
	{
	        this.mApplicationPartido.mButtonHit = 0;
		this.mApplicationPartido.setGame(new GamePartido(this.mApplicationPartido));
		this.mApplicationPartido.getGame().createStates();
		this.mApplicationPartido.getGame().setStates();
 		this.mApplicationPartido.sendJoinGame('2');
                this.mApplicationPartido.mStateMachine.changeState(this.mApplicationPartido.mApplicationPlay);
	}
},

exit: function()
{
	this.parent();
}

});
